core/script_language.cpp

```diff
@@ -355,6 +355,38 @@ ScriptCodeCompletionCache::ScriptCodeCompletionCache() {
 	singleton = this;
 }
 
+void ScriptLanguage::get_core_type_words(List<String> *p_core_type_words) const {
```

Comment:
This could likely be used to remove the hardcoded core type names from `modules/gdscript/gdscript_editor.cpp`.

That's likely work for another PR though. There might be more synergy possible between the syntax highlighter code and the auto-completion code, at least as far as identifying core types, classes and keywords goes. (Or maybe not, it has to be assessed whether it makes sense/would be convenient or would just be too cumbersome.)